packages/taquito/src/prepare/prepare-provider.ts

```diff
@@ -1098,4 +1101,75 @@ export class PrepareProvider extends Provider implements PreparationProvider {
       contents,
     };
   }
+
+  /**
+   *
+   * @description Method to prepare a stake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  stake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'stake');
+  }
+
+  /**
+   *
+   * @description Method to prepare an unstake operation
+   * @param params RPCOperation object
+   * @param source string or undefined source pkh
+   * @returns a PreparedOperation object
+   */
+  async unstake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'unstake');
+  }
+
+  /**
+   *
+   * @description Method to prepare a finalize_unstake operation
+   * @param params RPCOperation object
+   * @returns a PreparedOperation object
+   */
+  async finalizeUnstake(params: StakingParams): Promise<PreparedOperation> {
+    return this.staking(params, 'finalize_unstake');
+  }
+
+  private async staking(
```

Comment:
think this function can be public and called directly from, contract-provider and estimate-provider

Reply:
May be a matter of taste, but I prefer the more encapsulated approach

Reply:
@dsawali ?